plugins/outputs/zabbix/lld.go

```diff
@@ -0,0 +1,227 @@
+package zabbix
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"hash/fnv"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/metric"
+)
+
+const (
+	lldName = "lld"
+	empty   = `{"data":[]}`
+)
+
+type lldInfo struct {
+	Hostname string
+	Key      string
+	DataHash uint64
+	Data     map[uint64]map[string]string
+}
+
+func (i *lldInfo) hash() uint64 {
+	ids := make([]uint64, 0, len(i.Data))
+	for id := range i.Data {
+		ids = append(ids, id)
+	}
+	sort.Slice(ids, func(i, j int) bool { return ids[i] < ids[j] })
+
+	h := fnv.New64a()
+	// Write cannot fail
+	_ = binary.Write(h, internal.HostEndianness, lldSeriesID(i.Hostname, i.Key))
+	_, _ = h.Write([]byte{0})
+	_ = binary.Write(h, internal.HostEndianness, ids)
```

Comment:
I don't think this is still needed. I think the linters are updated. Right @srebhan?

```suggestion
	binary.Write(h, internal.HostEndianness, lldSeriesID(i.Hostname, i.Key))
	h.Write([]byte{0})
	binary.Write(h, internal.HostEndianness, ids)
```

Reply:
I have done a rebase and executed `make lint` and with those changes it fails:
```
golangci-lint run
plugins/outputs/zabbix/lld.go:39:14  errcheck  Error return value of `binary.Write` is not checked
plugins/outputs/zabbix/lld.go:41:14  errcheck  Error return value of `binary.Write` is not checked
```

Reply:
Indeed, the respective PRs are not merged yet.

Reply:
I think we cannot whitelist `binary.Write` as it might write to any stream, e.g. a network socket, and thus *might* error. So please keep the former as it cannot error for hashes...